contracts/src/v0.8/ccip/pools/GHO/UpgradeableLockReleaseTokenPool.sol

```diff
@@ -0,0 +1,249 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import {Initializable} from "solidity-utils/contracts/transparent-proxy/Initializable.sol";
+
+import {ILiquidityContainer} from "../../../liquiditymanager/interfaces/ILiquidityContainer.sol";
+import {ITypeAndVersion} from "../../../shared/interfaces/ITypeAndVersion.sol";
+
+import {IERC20} from "../../../vendor/openzeppelin-solidity/v4.8.3/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "../../../vendor/openzeppelin-solidity/v4.8.3/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC165} from "../../../vendor/openzeppelin-solidity/v5.0.2/contracts/utils/introspection/IERC165.sol";
+
+import {Pool} from "../../libraries/Pool.sol";
+import {IRouter} from "../../interfaces/IRouter.sol";
+import {UpgradeableTokenPool} from "./UpgradeableTokenPool.sol";
+
+/// @title UpgradeableLockReleaseTokenPool
+/// @author Aave Labs
+/// @notice Upgradeable version of Chainlink's CCIP LockReleaseTokenPool
+/// @dev Contract adaptations:
+/// - Implementation of Initializable to allow upgrades
+/// - Move of allowlist and router definition to initialization stage
+/// - Addition of a bridge limit to regulate the maximum amount of tokens that can be transferred out (burned/locked)
+/// - Permissioned bridgedAmount setter to facilitate liquidity migration
+/// - Remove i_token decimal check in UpgradeableTokenPool constructor
```

Comment:
same as in `UpgradeableBurnMintTokenPool`

Reply:
[4e0e90f](https://github.com/aave/ccip/pull/20/commits/4e0e90fbded84e592affeea4dacf5e597f6a251b)